src/firewall.py

```diff
@@ -66,7 +66,25 @@ def get_host_ip(self) -> str:
         )
         return str(ipaddress.IPv4Interface(address.strip()).ip)
 
-    def refresh_firewall(self, denylist: typing.List[FirewallEntry]):
+    def _exclude_network(
+        self,
+        networks: typing.List[ipaddress.IPv4Network],
+        excludes: typing.List[ipaddress.IPv4Network],
+    ) -> typing.Set[ipaddress.IPv4Network]:
+        total: typing.Set[ipaddress.IPv4Network] = set()
+        for exclude in excludes:
+            for network in networks:
+                try:
+                    total.update(network.address_exclude(exclude))
+                except ValueError:
+                    total.update([network])
+        return total  # this currently doesn't work
```

Comment:
Could you explain what you mean by that comment?

Reply:
Sorry it was a comment for me before I went to sleep - removed, thank you!